carp.client.core/src/commonTest/kotlin/dk/cachet/carp/client/application/ClientManagerTest.kt

```diff
@@ -0,0 +1,78 @@
+package dk.cachet.carp.client.application
+
+import dk.cachet.carp.client.domain.*
+import dk.cachet.carp.common.UUID
+import kotlin.test.*
+
+
+/**
+ * Tests for [ClientManager].
+ */
+class ClientManagerTest
+{
+    @Test
+    fun add_study_succeeds()
+    {
+        // Create deployment and client manager.
+        val ( deploymentManager, deploymentStatus) = createStudyDeployment( createSmartphoneStudy() )
+        val clientManager = SmartphoneManager( smartphone.createRegistration(), deploymentManager )
```

Comment:
Wouldn't it be better to get the `smartphone` from the `deploymentStatus`?
Then who ever are going to consume the client subsystem, will now how to initialize the `SmartphoneManager` based on information in the `deploymentStatus`.

Reply:
In a normal use case you would retrieve it through `StudyDeploymentStatus`. However, for the unit tests here I use a hardcoded object to reduce the LOC of each test (this is quite repetitive).

Reply:
Oh, I am mistaken. I simply used `smartphone.createRegistration` here to simply creating the matching `DeviceRegistration`. I think what would be used in practice is `DefaultDeviceRegistrationBuilder().build()`, for which perhaps a cleaner API needs to be provided.

Reply:
I do get you want to reduce the LOC for each test.
But couldn't you create a shared function to retreive the smartphone from the `deploymentStatus`?